dds/DCPS/DataWriterImpl_T.h

```diff
@@ -332,6 +332,17 @@ DataWriterImpl_T
         Encoding::Kind encoding_kind;
         if (repr_to_encoding_kind(repIds[i], encoding_kind)) {
           encoding_mode_ = EncodingMode(encoding_kind, swap_bytes());
+            if (encoding_kind == Encoding::KIND_XCDR1) {
+              if (MarshalTraitsType::max_extensibility_level() == MUTABLE) {
```

Comment:
```suggestion
            if (encoding_kind == Encoding::KIND_XCDR1 && MarshalTraitsType::max_extensibility_level() == MUTABLE) {
```

Same thing about the indentation.